src/Microsoft.Health.Dicom.Core/Features/Diagnostic/LogForwarderExtensions.cs

```diff
@@ -44,4 +48,44 @@ internal static class LogForwarderExtensions
 
         telemetryClient.TrackTrace(telemetry);
     }
+
+    /// <summary>
+    /// Emits a trace log with forwarding flag set for operations and adds the required properties to telemetry.
+    /// </summary>
+    /// <param name="telemetryClient">client to use to emit the trace</param>
+    /// <param name="message">message to set on the trace log</param>
+    /// <param name="operationId">operation id</param>
+    /// <param name="input">Input payload to pass to the forward logger</param>
+    public static void ForwardOperationLogTrace(
+        this TelemetryClient telemetryClient,
+        string message,
+        string operationId,
+        string input)
+    {
+        EnsureArg.IsNotNull(telemetryClient, nameof(telemetryClient));
+        EnsureArg.IsNotNull(message, nameof(message));
+
+        // Shoebox property size has a limitation of 32 KB which is why the diagnostic log is split into multiple messages
+        int startIndex = 0, offset = 0, inputSize = input.Length;
+        while (startIndex < inputSize)
+        {
+            offset = Math.Min(MaxShoeboxPropertySize, input.Length - startIndex);
+            ForwardOperationLogTraceWithSizeLimit(telemetryClient, message, operationId, input.Substring(startIndex, offset));
```

Comment:
What is the usability of this? lol
A bunch of copies of the log with partial inputs? Is there an alternative?

Reply:
There is a 32KB limitation for the property added, which is why we split the input message into multiple diagnostic messages. We discussed multiple approaches like either truncate anything after 32KB or split the parameters and add them separately. But the one we are currently doing seems to be the better approach.